src/Application.php

```diff
@@ -639,33 +469,9 @@ private function checkForDuplicateRoute($path, $methods = null)
      */
     private function emitMarshalServerRequestException($exception)
     {
-        $response = (new Response())
+        $response = $this->responsePrototype
             ->withStatus(StatusCode::STATUS_BAD_REQUEST);
-        $finalHandler = $this->getFinalHandler();
-        $response = $finalHandler(new ServerRequest(), $response, $exception);
```

Comment:
Yes, so maybe I've placed comment on wrong place, please see on signature of method `emitMarshalServerRequestException` - there is parameter `$exception, which now is never used, because above line is removed. We should then remove this parameter in the method signature and all invocations all do something with the exception here.